### PR TITLE
Adding CAPC templates

### PR DESCRIPTION
**Pull request by maxdrib**
*Issue #, if available:*

*Description of changes:*
As requested per https://github.com/aws/eks-anywhere/pull/1299/files, splitting out the CAPC templates into a separate PR for analysis before adding the cloudstack provider code

*Testing (if applicable):*
Tested in conjunction with https://github.com/aws/eks-anywhere/pull/1299/files and successfully created a cluster

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/providers/cloudstack/config/template-cp.yaml [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    spec:
+      offering: {{.cloudstackEtcdComputeOffering}}
+      template: {{.cloudstackEtcdTemplateOffering}}
+{{- if .cloudstackEtcdAffinityGroupIds }}
+      affinitygroupids:
+      {{- range .cloudstackEtcdAffinityGroupIds }}
+      - {{.}}
+      {{- end }}
+{{- end }}
+---
```
Comment (vivek-koppuru) [inline pkg/providers/cloudstack/config/template-cp.yaml:304]: Is this extra?

File: pkg/providers/cloudstack/config/template-cp.yaml [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+    users:
+    - name: {{.controlPlaneSshUsername}}
+      sshAuthorizedKeys:
+      - '{{.cloudstackControlPlaneSshAuthorizedKey}}'
+      sudo: ALL=(ALL) NOPASSWD:ALL
+    format: {{.format}}
+  replicas: {{.controlPlaneReplicas}}
+  version: {{.kubernetesVersion}}
+---
+{{- if .externalEtcd }}
```
Comment (vivek-koppuru) [inline pkg/providers/cloudstack/config/template-cp.yaml:241]:
```suggestion
{{- if .externalEtcd }}
---
```
Similar to the comment below, but do we need these trailing delimiters if our templater will add them anyways?

Comment (maxdrib) [reply pkg/providers/cloudstack/config/template-cp.yaml:241]: I'm not sure if I follow. I was under the impression the templater just evaluates the logic and fills in variable as specified. Are you concerned about having duplicate delimiters, such as in the case where .exteralEtcd is undefined?

Comment (vivek-koppuru) [reply pkg/providers/cloudstack/config/template-cp.yaml:241]: Yup mainly about duplicate delimiters, which might be a problem in vsphere too. Might be better to catch in the review that we have here, which was why I listed it as a comment

File: pkg/providers/cloudstack/config/template-cp.yaml (hunk 1 above)
Comment (maxdrib) [reply pkg/providers/cloudstack/config/template-cp.yaml]: removed

File: pkg/providers/cloudstack/config/template-cp.yaml (hunk 2 above)
Comment (maxdrib) [reply pkg/providers/cloudstack/config/template-cp.yaml]: Refactored :)

**Review by vivek-koppuru — APPROVED** [review]
/lgtm

**Pull request merged**